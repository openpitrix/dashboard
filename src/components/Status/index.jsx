import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { capitalize } from 'lodash';
import { translate } from 'react-i18next';

import styles from './index.scss';

@translate()
export default class Status extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    name: PropTypes.string,
    style: PropTypes.string,
    transition: PropTypes.oneOf([
      '',
      'starting',
      'updating',
      'stopping',
      'deleting',
      'creating',
      'upgrading',
      'rollbacking',
      'recovering',
      'ceasing',
      'resizing',
      'scaling'
    ]),
    /* type: PropTypes.oneOf([
     *   'draft',
     *   'running',
     *   'active',
     *   'stopped',
     *   'ceased',
     *   'pending',
     *   'suspended',
     *   'successful',
     *   'failed',
     *   'deleted',
     *   'working', // job status

     *   'enabled',
     *   'disabled',

     *   'published'
     * ]), */
    type: PropTypes.string
  };

  static defaultProps = {
    type: 'pending',
    transition: ''
  };

  render() {
    const {
      style, className, name, type, transition, t
    } = this.props;
    const status = String(transition || type).toLowerCase();

    const normalizeName = t(capitalize(name || status));

    return (
      <span className={classnames(styles.status, className)} style={style}>
        <i className={classnames(styles.icon, styles[status])} />
        {normalizeName}
      </span>
    );
  }
}
