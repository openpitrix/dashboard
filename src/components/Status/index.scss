@import '~scss/vars';

.status {
  text-transform: capitalize;
  display: inline-flex;
  flex-direction: row;
  height: 100%;
  align-items: center;

  .name {
    margin-left: 4px;
  }
}

.icon {
  display: inline-block;
  vertical-align: middle;
  width: 12px;
  height: 12px;
  margin: 0 4px 2px 0;
  border-radius: 50%;
  font-style: normal;

  &.draft,
  &.submitted {
    background-color: $B75;
  }
  &.in-review,
  &.isv-in-review,
  &.business-in-review,
  &.dev-in-review {
    background-color: $Y300;
  }
  &.published,
  &.passed,
  &.isv-passed,
  &.business-passed,
  &.dev-passed {
    background-color: $G75;
  }
  &.rejected,
  &.isv-rejected,
  &.business-rejected,
  &.dev-rejected {
    background-color: $R200;
  }
  &.suspended {
    background-color: #343945;
  }
  &.ceased {
    background-color: $Y100;
  }
  &.stopped {
    background-color: $N30;
  }
  &.active,
  &.successful {
    background-color: $G75;
  }
  &.deleted,
  &.failed {
    background-color: $R300;
  }
  &.enabled {
    background-color: $G65;
  }
  &.disabled {
    background-color: $N75;
  }

  &.running,
  &.working,
  &.starting,
  &.pending,
  &.updating,
  &.stopping,
  &.upgrading,
  &.creating,
  &.rollbacking,
  &.recovering,
  &.ceasing,
  &.resizing,
  &.scaling,
  &.deleting {
    @include transitionStatus;
  }

  &.starting,
  &.creating {
    background-color: $P75;
    border-color: $P65;
  }
  &.running,
  &.working {
    background-color: $G200;
    border-color: $G100;
  }
  &.updating,
  &.upgrading,
  &.resizing,
  &.scaling {
    background-color: $Y75;
    border-color: $Y65;
  }
  &.stopping,
  &.ceasing {
    background-color: $Y300;
    border-color: $Y200;
  }
  &.pending,
  &.rollbacking,
  &.recovering {
    background-color: $B75;
    border-color: $B65;
  }
  &.deleting {
    background-color: $R200;
    border-color: $R100;
  }
}
