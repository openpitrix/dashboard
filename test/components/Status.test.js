import Status from 'components/Status';

describe('Component Status', () => {
  it('basic render', () => {
    const wrapper = render(<Status name="active" type="active" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has transition', () => {
    const wrapper = render(<Status transition="starting" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
