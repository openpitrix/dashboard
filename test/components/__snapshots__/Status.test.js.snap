// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Status basic render 1`] = `
<span
  class="status"
>
  <i
    class="icon active"
  />
  <span
    class="name"
  >
    Active
  </span>
</span>
`;

exports[`Component Status has transition 1`] = `
<span
  class="status"
>
  <i
    class="icon starting"
  />
  <span
    class="name"
  >
    Starting
  </span>
</span>
`;
